dipy/reconst/mapmri.py

```diff
@@ -136,13 +142,19 @@ def __init__(self,
             optimal weight from the values in the array.
         positivity_constraint : bool,
             Constrain the propagator to be positive.
+        constraint_type : string,
+            If set to 'local', positivity is enforced on a grid determined by
```

Comment:
Are there only 2 constraint_types?  if yes, I would recommend using a boolean value instead of a string value. it avoids a lot of issues. your variable can be renamed `use_local_constraint=True`